pkg/user/pruning_test.go

```diff
@@ -0,0 +1,51 @@
+package user
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPruningInTxTracker(t *testing.T) {
+	txClient := &TxClient{
+		txTracker: make(map[string]txInfo),
+	}
+	numTransactions := 10
+
+	// Add 10 transactions to the tracker that are 10 minutes old
```

Comment:
this comment seems incorrect. It looks like the implementation below adds 5 transactions that are 10 minutes old and 5 transactions that are 5 minutes old.

```suggestion
```